graphql-server/src/utils/cache.ts

```diff
@@ -0,0 +1,218 @@
+import fetch from "node-fetch";
+import NodeCache from "node-cache";
+import { props } from "./constants";
+import { neo4jDriver } from "./servers";
+import {
+  buildSamplesQueryBody,
+  buildSamplesQueryFinal,
+  queryDashboardSamples,
+} from "../schemas/queries/samples";
+import { DashboardRecordSort, DashboardSample } from "../generated/graphql";
+
+export const ONCOTREE_CACHE_KEY = "oncotree";
+export const SAMPLES_CACHE_KEY = "samples";
+
+const ONCOTREE_CACHE_TTL = 86400; // 1 day
+const SAMPLES_CACHE_TTL = 3600; // 1 hour
+
+const SAMPLES_PAGES_TO_CACHE = 5; // lower this if needed during dev for faster iterations
+const CACHE_BLOCK_SIZE = 100; // keep consistent with helpers.tsx's CACHE_BLOCK_SIZE
+// Keep consistent with SamplesList.tsx's DEFAULT_SORT
+const SAMPLES_DEFAULT_SORT = {
+  colId: "importDate",
+  sort: "desc",
+} as DashboardRecordSort;
+// Keep consistent with SamplesPage.tsx's WES_SAMPLE_CONTEXT
+const WES_SAMPLE_CONTEXT = {
+  fieldName: "genePanel",
+  values: [
+    "Agilent_51MB",
+    "Agilent_v4_51MB_Human",
+    "CustomCapture",
+    "IDT_Exome_v1_FP",
+    "IDT_Exome_V1_IMPACT468",
+    "WES_Human",
+    "WholeExomeSequencing",
+  ],
+};
```

Comment:
This has me wondering if we can define a global constant in neo4j that `codegen` will automatically generate the react types for that we can use in the frontend/backend dashboard as needed similar to how we're already using the types we define for the `DashboardSample`, etc.

Reply:
Not a requirement, just a thought! Trying to think of ways that we can reduce the number of places we'd need to maintain consistency between the two areas of the code.